### PR TITLE
Experimental Mineral Balance PR.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

Good morning fokes, this PR is going to attempt to tweak around some of the values of mineral costs in-game in the hopes of making the mineral economy more engaging to the base gameplay loop, in the hopes that by making it more self-realized, it would allow for better integration with the rest of the gameplay loop as well as so that it doesn't completely overshadow the credit economy. That said, **this pr only makes changes to mined and station minerals**.

So, the base change is as follows: items with designated costs of minerals below 1000 minerals, as designated in #75052 as `SMALL_MATERIAL_AMOUNT`, have been left alone. Any item above 1000 or 2000, from `HALF_SHEET_MATERIAL_AMOUNT` and `SHEET_MATERIAL_AMOUNT` have been adjusted, as sheets will now contain **100 minerals each**.

For example, if an object costed 4000 iron, 1000 glass, and 500 diamond with the old numbers, with the changes to these defines, that item would cost 2000 iron, 500 glass, and 500 diamond.

## Why It's Good For The Game

Arcane has been doing some statistics now that he's back in the land of the living. So, I broke down a 40 round aggregate of silo logs, excluding rounds where the silo was left untouched (as, I can't imagine a round where nothing was both lathed or mined would be statistically relevant). These rounds had their minerals obtained tallied, and the amount spent and obtained were compared, giving a net total. The amount consumed as a percentage was then obtained as an average of all rounds, as a measurement of average "health" of that mineral within the material economy.

The results were as follows:
image

In the average round, iron and glass being the most consumed mineral makes sense, since the most useful and most bountiful items are typically iron/glass exclusives. Not to mention, if a tool has to be obtained later in the shift, it's either research locked or it has to be very, VERY good to warrant waiting for it. So that value makes sense. Iron is also typically the most in-demand mineral, as well as the most requested material as a result of having to fix hull breaches. Glass by comparison has issues on some maps like icebox, but overall it's useful enough that it has a much MUCH higher quantity obtained than other materials.

Other mined minerals typically don't break over 50 sheets (100,000 in this metric), and are also consumed less than 30% at maximum, and 14% at the lowest. This is for a number of reasons, but the big two are as follows:
* "Rare" resources are rarely taken in quantities greater than the previous SMALL_MATERIAL_AMOUNT on their most valuable parts, which means that despite the quantity that they come in at, there's no real scarcity of the resource, which makes the bulk of rare material acquisition meaningless after the first load of ores comes in.
* "Rare" resources don't have the same level of usefulness as iron/glass, as they are unique, those unique properties don't help the station actually function more effectively. Gold can't patch a hole the size of medbay, and silver cannot help to refill oxygen to a station doomed to recall. This is hyperbole, I know, but there's nothing that really drives needing specific rare materials due to them not having a truly unique identity within the game. IE: If high cost silver items were a driving force in making sterile workshops in medical, seeing an empty silver reserve becomes an indicator of medical department's state.

So, where am I going with all this?

This PR should serve as an experiment in order to see if by pushing the lower end of mineral consumption, usually taken up by "rare" minerals, we see a better percentage of mineral usage causing trade-offs in gameplay after the first 15 minutes of a shift. By normalizing regular sheets to a standard 1000 units, it also means that finally one unit of a sheet can be a "millisheet", as we've all been demanding for years (this is sarcasm but the uniformity is rather nice).

DNM until it's been testmerged and we can compare some logged rounds silo data, both from an LRP server as well as from MRP. (MRP rounds, being longer in general, gave much higher values while also showing a higher level of surplus, where LRP would be much shorter logs by far, but could have a much wider range of values due to the volatile nature of rounds ending earlier than expected).

## Changelog

:cl:
balance: In general, mineral recipe costs for items in larger amounts has been decreased by 2x.
balance: One sheet now contains a total of 100 units of material, down from 2000. Proportionally the costs of items should remain the same, except for the smaller costs of certain recipes. 
code: Added a python script to the tools folder, letting you see a mineral breakdown of one/several rounds using silo logs.
/:cl:

### CODE REVIEW

File: code/__DEFINES/construction.dm [hunk 1]
```diff
@@ -75,11 +75,11 @@
 
 // Defines related to the custom materials used on objects.
 ///The amount of materials you get from a sheet of mineral like iron/diamond/glass etc. 2000 Units.
```
Comment (JohnFulpWillard) [inline code/__DEFINES/construction.dm:77]: update the defines please

**Comment by CRITAWAKETS** [comment]
might want to look at tweaking roboticist material values with this, since practically everything roboticists make is defined with SHEET_MATERIAL_VALUE, meaning their materials consumption would be halved.

And we all know just how much materials roboticists use.

**Comment by atlasle** [comment]
in the changelog, "halved" would be clearer than "decreased by 2x"

**Comment by MTandi** [comment]
So now you can make less stock parts from 1 sheet, so items below 1k became more expensive. Did I get this wrong?

**Comment by YehnBeep** [comment]
So, question. Does this factor in resource usage over time?

Because, for example, early and sometimes mid round silver (and uranium) shortages are very much a thing at times. I'm worried that if this is only looking at end of round totals, that may be skewing results. i.e., a surplus when leaving at the end of the round may misrepresent supply and demand pressures as they're shaped through ongoing gameplay.

That being said, qualitatively, I rarely see complaints of mineral shortages outside of silver (that being the most non-metal/glass complaint by far), and then occasionally uranium and diamond (the last tending to come from engineering projects).

**Review by JohnFulpWillard — COMMENTED** [review]
can I suggest we go further than this

1000 -> 500 -> 100

100 -> 50 -> 10

It's way easier to look/deal with/calculate smaller numbers, and it's good game design to give players smaller numbers.

Unironically my favorite example is in LISA, which does not display health and all enemies have large HP pools, you just attack and attack and hope the enemies eventually die, compared to LISA the Pointless which has decreased everyone's HP and damage by 10x, where you know exactly what damage you do, you can know for sure whether you'll kill an enemy or not this turn.

**Comment by atlasle** [comment]
> can I suggest we go further than this
> 
> 1000 -> 500 -> 100
> 
> 100 -> 50 -> 10
> 
> It's way easier to look/deal with/calculate smaller numbers, and it's good game design to give players smaller numbers.
> 
> Unironically my favorite example is in LISA, which does not display health and all enemies have large HP pools, you just attack and attack and hope the enemies eventually die, compared to LISA the Pointless which has decreased everyone's HP and damage by 10x, where you know exactly what damage you do, you can know for sure whether you'll kill an enemy or not this turn.

LISA: The Pointless is the best, thank you John (gummy worms will be added soon)

**Comment by ArcaneMusic** [comment]
Howdy everyone, and good evening. This is your minerals and rocks weekly update, by yours truely.

So, some quick observations and alterations to strategy tonight:
For starters, I am a massive dum dum and may have potentially divided the wrong two numbers in coming up with my mineral percentages used, which has been summarily corrected.
image
That said, here's the aggregate of the old numbers by comparison to the new numbers. By and large, we do see a higher percentage of minerals across the board being used when it comes to more "common" minerals. This is expected, and not necessarily all that bad.

What is interesting is that for the most part, our "rare" minerals are within standard deviation of where they were before, despite a similar sample size and similar vetting (IE: not including rounds where the ore silo goes untouched). I can't say this is what I was hoping to see, as that means there'll need to be a more nuanced approach towards mineral quantities and costs past just increasing the small costs as they appear. But, alas.

Interesting thing to note that was called to my attention, so the ore silo log covers ores deposited into (proto)lathes, but not (auto)lathes, since they live in their own isolated network. Technically, they are covered by "released" minerals being taken back out of the ORM and back into the lathe, but that doesn't tell us how much of those materials are being used meaningfully, since that was... kinda part of the point. 

I'm very down for @JohnFulpWillard 's thoughts on bringing the sheet defines down, if only just to make mineral counts look less like a keen mess for players trying to read into mineral amounts, but I do think the proposed 100->50->10 is about as low as granularity may possibly allow for certain lathe recipes (in my ideal world decimal minerals wouldn't exist in the ORM).

Past that, I'll take off that pesky self-DNM label as I think I've hit when I can for now.

**Comment by SyncIt21** [comment]
https://github.com/tgstation/tgstation/blob/d25e096e13188fec6f412c28951601bdbe86f450/code/modules/research/designs/stock_parts_designs.dm#L233

`/datum/material/diamond = 60`

amount of diamond is still a magic number?

**Comment by ArcaneMusic** [comment]
> Scroll full right
> 
> https://github.com/tgstation/tgstation/blob/d25e096e13188fec6f412c28951601bdbe86f450/code/modules/research/designs/stock_parts_designs.dm#L233
> 
> `/datum/material/diamond = 60`
> 
> amount of diamond is still a magic number?

Good catch, easy to miss one or two of these

**Comment by YehnBeep** [comment]
> What is interesting is that for the most part, our "rare" minerals are within standard deviation of where they were before, despite a similar sample size and similar vetting (IE: not including rounds where the ore silo goes untouched). I can't say this is what I was hoping to see, as that means there'll need to be a more nuanced approach towards mineral quantities and costs past just increasing the small costs as they appear. But, alas.

Qualitatively, for what it's worth, I've seen an increase in pressure from the science department on miners, and things like the HoP doubling the miner slots and marking it as a priority job.

I still have my concerns about your methodology. I don't think looking at round-end amounts reflects the impacts on play as the round progresses.

**Comment by Breadeagle** [comment]
No offense, but I think there's some major problems with the methodology if you're aiming to use statistics to form a conclusion here. 

To start, I don't see any distinction between Icebox and Lavaland - if that was divided when you scraped your data, then it wasn't made clear. Icebox is vastly more plentiful between the actual ore density, lack of shuttle requirement, 3 mineable levels, and easy public mining. It also has more rewarding mobs, including one that drops bluespace crystals. Its not uncommon to see a "finished" ore silo by 15-20 minutes.

The other problem, as already mentioned here, is that you only seem to be looking at end of round totals, which is an extremely narrow set of data to view. If you're looking to learn about the mid-round economy, these stats tell you effectively nothing. If there's a deficit at any point in the round, its not made clear by end of round stats. There's also an absence of a detailed breakdown on what materials are spent on exactly. To make a truly informed decision using mostly statistics, we'd need probably several orders of magnitude more data. 

I'm not convinced simply adjusting the price of current items will ever make the ore economy "engaging" since the bluespace silo exists, and the need to physically retrieve or deliver the materials doesn't exist anymore, either. The more ideal fix, in the end, is probably to have more items that people actually want to make. From my experience, the most produced items are usually miner tools, stock parts, and advanced engi/med tools. Over the years a few items that were commonly produced became more restricted, such as guns. You talk about a hypothetical where medical might need a lot of silver for upgrades (maybe like Rimworld), and I'm not sure I can see that feeling satisfying given that anybody with lathe access can directly pull out all of a resource for the whole station. I know that's just an example, and not 100% related to this exact PR, anyway.

Don't really have any objections to changing sheet quantity though. I'd place my support for 1000, probably.

TL;DR: There's some problems with the stats interpretation, and some other thoughts.

**Comment by Breadeagle** [comment]
Oh yeah, one more thing that I wasn't sure where to put. For common materials, one of the reasons their consumption is so high is that their rewards for miners are _extremely_ low. For iron its often literally not possible to get enough for the station via lavaland mining alone. For glass, the quantity isn't an issue but they're worth basically no points, the same as iron. Mining either of these is often a "waste of time" for a selfish miner, and for those wanting to help the station, cargo orders are significantly faster and easier. Anyway, the reason consumption is so high is because of this, and also cargo players seeming to not order materials very often. I see players yell at miners to get iron and glass, but they have no really efficient method to actually do so. 

Sorry if that's too off topic - saw you compare the common to rare materials and this came to mind.

**Comment by themicrowave1** [comment]
Noticed the service cyborg's mods weren't working with this system, making them astronomically expensive.

**Comment by Jackal-boop** [comment]
miners seem to get less points now because of this

**Comment by SyncIt21** [comment]
Also not all item's prices/custom material's were adjusted to use the correct Defines. Some things are way too expensive than they need to be #75554

**Comment by carlarctg** [comment]
Not sure if intended but I noticed advanced health analyzers costed 5k glass and normal health analyzers only like 500

**Comment by ZAMODA** [comment]
Before this PR hardly anyone would complain about shortages, but due to the fact that lathes were not properly updated along with these changes, I have personally seen a major increase in people complaining about materials. (also it takes HALF the stations supply of iron to make **6** signalers)

**Comment by ArcaneMusic** [comment]
I was away all weekend, so I'll fix the defines ASAP. TM has been stopped for now.

**Comment by ArcaneMusic** [comment]
> > Scroll full right
> > https://github.com/tgstation/tgstation/blob/d25e096e13188fec6f412c28951601bdbe86f450/code/modules/research/designs/stock_parts_designs.dm#L233
> > 
> > `/datum/material/diamond = 60`
> > amount of diamond is still a magic number?
> 
> Good catch, easy to miss one or two of these

I stand corrected, I missed a hell of a lot of defines. I'm starting on these now but because I have more travel ahead of me because of work, I'll see if we can get this worked through slowly but surely.

**Comment by JohnFulpWillard** [comment]
Items that don't use defines
``/obj/item/autopsy_scanner`` & ``/datum/design/autopsy_scanner``
``/datum/design/golem_shell``
``/datum/design/board/freeformcore_module``
``/datum/design/beacon``
``/datum/design/bag_holding``
``/datum/design/miningsatchel_holding``
``/datum/design/portabledrive/basic``
``/datum/design/portabledrive/advanced``
``/datum/design/portabledrive/super``
``/datum/design/ai_cam_upgrade``
``/datum/design/phazon_torso``
``/datum/design/borg_upgrade_restart``
``/datum/design/borg_upgrade_disablercooler``
``/datum/design/borg_upgrade_rolling_table``
``/datum/design/borg_upgrade_condiment_synthesizer``
``/datum/design/borg_upgrade_silicon_knife``
``/datum/design/borg_upgrade_service_apparatus``
``/datum/design/boris_ai_controller``
``/datum/design/plasmarefiller``
``/datum/design/defibrillator``
``/datum/design/defibrillator_compact``
``/datum/design/healthanalyzer_advanced``
``/datum/design/medigel``
``/datum/design/laserscalpel``
``/datum/design/mechanicalpinches``
``/datum/design/searingtool``
``/datum/design/portable_chem_mixer``
``/datum/design/cyberimp_welding``
``/datum/design/cyberimp_gloweyes``
``/datum/design/cyberimp_breather``
``/datum/design/cyberimp_medical_hud``
``/datum/design/cyberimp_security_hud``
``/datum/design/cyberimp_diagnostic_hud``
``/datum/design/cyberimp_xray``
``/datum/design/cyberimp_thermals``
``/datum/design/cyberimp_antidrop``
``/datum/design/cyberimp_antistun``
``/datum/design/cyberimp_nutriment_plus``
``/datum/design/cyberimp_reviver``
``/datum/design/cyberimp_thrusters``
``/datum/design/implanter``
``/datum/design/implant_chem``
``/datum/design/cybernetic_ears``
``/datum/design/cybernetic_eyes``
``/datum/design/drill``
``/datum/design/drill_diamond``
``/datum/design/plasmacutter``
``/datum/design/jackhammer``
``/datum/design/superresonator``
``/datum/design/hyperaccelerator``
``/datum/design/health_hud_night``
``/datum/design/security_hud_night``
``/datum/design/diagnostic_hud_night``
``/datum/design/bright_helmet``
``/datum/design/rolling_table``
``/datum/design/portaseeder``
``/datum/design/air_horn``
``/datum/design/nvgmesons``
``/datum/design/night_vision_goggles``
``/datum/design/magboots``
``/datum/design/forcefield_projector``
``/datum/design/nv_sci_goggles``
``/datum/design/roastingstick``
``/datum/design/donksoft_refill``
``/datum/design/pneumatic_seal``
``/datum/design/advmop``
``/datum/design/adv_watering_can``
``/datum/design/reactive_armour``
``/datum/design/knight_armour``
``/datum/design/knight_helmet``
``/datum/design/seclite``
``/datum/design/zipties``
``/datum/design/plumbing_rcd``
``/datum/design/tackle_dolphin``
``/datum/design/alienalloy``
``/datum/design/rped``
``/datum/design/bs_rped``
``/datum/design/adv_capacitor``
``/datum/design/phasic_scanning``
``/datum/design/adv_scanning``
``/datum/design/triphasic_scanning``
``/datum/design/nano_servo``
``/datum/design/femto_servo``
``/datum/design/high_micro_laser``
``/datum/design/ultra_micro_laser``
``/datum/design/quadultra_micro_laser``
``/datum/design/adv_matter_bin``
``/datum/design/bluespace_matter_bin``
``/datum/design/subspace_crystal``
``/datum/design/card_reader``
``/datum/design/usb_cable``
``/datum/design/controller_shell``
``/datum/design/scanner_shell``
``/datum/design/keyboard_shell``
``/datum/design/gun_shell``
``/datum/design/bot_shell``
``/datum/design/money_bot_shell``
``/datum/design/drone_shell``
``/datum/design/server_shell``
``/datum/design/airlock_shell``
``/datum/design/dispenser_shell``
``/datum/design/bci_shell``
``/datum/design/scanner_gate_shell``
``/datum/design/solar``
``/datum/design/geiger``
``/datum/design/turret_control_frame``
``/datum/design/large_welding_tool``
``/datum/design/camera_assembly``
``/datum/design/newscaster_frame``
``/datum/design/status_display_frame``
``/datum/design/intercom_frame``
``/datum/design/extinguisher``
``/datum/design/pocketfireextinguisher``
``/datum/design/tscanner``
``/datum/design/requests_console``
``/datum/design/light_switch_frame``
``/datum/design/telescreen_turbine``
``/datum/design/telescreen_engine``
``/datum/design/telescreen_auxbase``
``/datum/design/rwd``
``/datum/design/analyzer``
``/datum/design/weldingtool``
``/datum/design/mini_weldingtool``
``/datum/design/screwdriver``
``/datum/design/wirecutters``
``/datum/design/wrench``
``/datum/design/plunger``
``/datum/design/welding_helmet``
``/datum/design/cable_coil``
``/datum/design/emergency_oxygen_engi``
``/datum/design/plasmaman_tank_belt``
``/datum/design/boxcutter``
``/datum/design/foilhat``
``/datum/design/large_beaker``
``/datum/design/condenser``
``/datum/design/signaler``
``/datum/design/health_sensor``
``/datum/design/prox_sensor``
``/datum/design/cleaver``
``/datum/design/desttagger``
``/datum/design/razor``
``/datum/design/conveyor_switch``
``/datum/design/laptop``
``/datum/design/paper_biscuit_confidential``
``/datum/design/rubbershot``
``/datum/design/c38``
``/datum/design/recorder``
``/datum/design/tape``
``/datum/design/electropack``
``/datum/design/receiver``
``/datum/design/shotgun_dart``
``/datum/design/incendiary_slug``
``/datum/design/riot_darts``
``/datum/design/a357``
``/datum/design/a762surplus``
``/datum/design/telescreen_interrogation``
``/datum/design/telescreen_prison``
``/datum/design/camera_film``
``/datum/design/kitchen_knife``
``/datum/design/fork``
``/datum/design/plastic_fork``
``/datum/design/spoon``
``/datum/design/plastic_spoon``
``/datum/design/plate``
``/datum/design/plant_analyzer``
``/datum/design/hatchet``
``/datum/design/secateurs``
``/datum/design/radio_headset``
``/datum/design/bounced_radio``
``/datum/design/handlabeler``
``/datum/design/pet_carrier``
``/datum/design/capbox``
``/datum/design/toy_balloon``
``/datum/design/plastic_tree``
``/datum/design/plastic_ring``
``/datum/design/chisel``
``/datum/design/toner``
``/datum/design/toner/large``
``/datum/design/ticket_machine``
``/datum/design/telescreen_bar``
``/datum/design/telescreen_entertainment``

all biogenerator designs (probably intentional)

**Comment by Gamer025** [comment]
Since people have been wondering about material usage over the course of a round, have this Python script that graphs every resources amount over the course of a round inside the silo neatly:
```
import json
import os
import pandas as pd
import plotly.express as px
import plotly.graph_objs as go

events = []
material_totals = {
    "iron": 0,
    "glass": 0,
    "silver": 0,
    "gold": 0,
    "uranium": 0,
    "titanium": 0,
    "bluespace crystal": 0,
    "diamond": 0,
    "plasma": 0,
    "bananium": 0,
    "plastic": 0,
}

events = []

log_folder = "logs"
current_file = 0

for filename in os.listdir(log_folder):
    with open(os.path.join(log_folder, filename), "r") as file:
        events.insert(current_file, {
            "iron": [],
            "glass": [],
            "silver": [],
            "gold": [],
            "uranium": [],
            "titanium": [],
            "bluespace crystal": [],
            "diamond": [],
            "plasma": [],
            "bananium": [],
            "plastic": [],
        })
        for line in file:
            log = json.loads(line)
            if (type(log) is dict and log.get("cat") == "silo"):
                for key, value in material_totals.items():
                    material_totals[key] = 0
                    events[current_file][key].append({"timestamp":log.get("ts"), "total":0})
                continue
            raw_materials = log[3].get("raw_materials", "")
            if not raw_materials:
                continue
            for raw_material in raw_materials.split(", "):
                quantity, material = raw_material.split(" ", 1)
                material_totals[material] += int(quantity)
                events[current_file][material].append({"timestamp":log[4].get("timestamp"), "total":int(material_totals[material])})
        current_file += 1

for round in events:
  dataframes = {}
  for key, value in round.items():
    df = pd.DataFrame(data=round[key])
    df["timestamp"] = pd.to_datetime(df["timestamp"], unit='s')
    df.index = df["timestamp"]
    df.sort_index()
    dataframes[key] = df
  fig = go.Figure()
  for i in dataframes:
      fig = fig.add_trace(go.Scatter(x = dataframes[i]["timestamp"],
                                    y = dataframes[i]["total"], 
                                    name = i))
  fig.update_layout(title = pd.to_datetime(round["iron"][0]["timestamp"], unit='s').strftime('%Y-%m-%d %X'))
  fig.show()
```
Just make sure to install the necessary modules with pip (json, plotly, pandas) and drop all your silo.json logs in a folder called "logs" next to the script (or change the path in the code)
For every log file a browser windows with interactive graph will open looking like this (recent Terry round):
image

**Comment by SyncIt21** [comment]
There is a problem with items that cost SMALL_MATERIAL_AMOUNT

Take the tier 4 micro laser for example

https://github.com/tgstation/tgstation/blob/d25e096e13188fec6f412c28951601bdbe86f450/code/modules/research/designs/stock_parts_designs.dm#L233

It's uranium cost for example is
`/datum/material/uranium =SMALL_MATERIAL_AMOUNT`

Let's compare the current and new values

**Current Values**
```
SHEET_MATERIAL_AMOUNT = 2000
SMALL_MATERIAL_AMOUNT = 100
```

So this means if i mine one ore of uranium(i.e SHEET_MATERIAL_AMOUNT amount of uranium) i will be able to make

`SHEET_MATERIAL_AMOUNT/SMALL_MATERIAL_AMOUNT = 2000/100 = 20 T4 lasers`

So the current production amount = 20

`New Values`

```
SHEET_MATERIAL_AMOUNT = 100
SMALL_MATERIAL_AMOUNT = 10
```
Doing the same math we get 

`SHEET_MATERIAL_AMOUNT/SMALL_MATERIAL_AMOUNT = 100/10 = 10 T4 lasers`

So the new production amount = 10

**The Conclusion?**
Current production amount =` 20 T4 lasers`
New Production amount = `10 T4 lasers`

You have effectively HALFED the production of micro laser's per sheet of uranium for no good reason so while your statement

> "Rare" resources are rarely taken in quantities greater than the previous SMALL_MATERIAL_AMOUNT on their most valuable parts, which means that despite the quantity that they come in at, there's no real scarcity of the resource, which makes the bulk of rare material acquisition meaningless after the first load of ores comes in

is currently true, this PR will now create that scarcity because all item's that cost SMALL_MATERIAL_AMOUNT will have their production amount halved unless miner's work twice as hard as before to keep the production amount same of those item's as before.(Good luck explaining to them why that's good thing)

**The Solution**

The current values are 

2000 -> 1000 -> 100

i.e. the ratio 1:2:10 

So you will have to keep this ration going forward which means the new values should be

100 -> 50 -> 5(not 10)

While yes it doesn't look uniform or nice it's the only way to keep your promise 

`in #75052 as SMALL_MATERIAL_AMOUNT, have been left alone`

Else we need to explain to everyone why there is a sudden scarcity of items that cost SMALL_MATERIAL_AMOUNT because miner's are unable to cope with their demand now

**Comment by YehnBeep** [comment]
> is currently true, this PR will now create that scarcity because all item's that cost SMALL_MATERIAL_AMOUNT will have their production amount halved unless miner's work twice as hard as before to keep the production amount same of those item's as before.(Good luck explaining to them why that's good thing)

Um... Not to be too pointed about it, but did you read the PR's body? At least my takeaway from it was that this is entirely the point. (The changing the scale of the numbers as well came after).

Now I will say I agree the impact is greater than the original's extremely coarse methodology/application of statistics would suggest (Thank you @Gamer025 for giving a higher resolution tool). Whether that's good or bad... well, I'll leave that to others to debate.

Anyways, the cost of less-than-a-sheet items effectively doubles under this PR, and this not an accident in the slightest.

**Comment by SyncIt21** [comment]
The PR body doesn't explain why that's a good thing. Like what is good about halfing production?

**Comment by Gamer025** [comment]
Made a second version of my script on request of @ArcaneMusic that now generates a single graph for every material type.
This is useful if you have lots of silo logs and want to see how different rounds behaved on ore consumption.
```
import json
import os
import pandas as pd
import plotly.express as px
import plotly.graph_objs as go

events = []
material_totals = {
    "iron": 0,
    "glass": 0,
    "silver": 0,
    "gold": 0,
    "uranium": 0,
    "titanium": 0,
    "bluespace crystal": 0,
    "diamond": 0,
    "plasma": 0,
    "bananium": 0,
    "plastic": 0,
}

events = []

log_folder = "logs"
current_file = 0

for filename in os.listdir(log_folder):
    with open(os.path.join(log_folder, filename), "r") as file:
        events.insert(current_file, {
            "iron": [],
            "glass": [],
            "silver": [],
            "gold": [],
            "uranium": [],
            "titanium": [],
            "bluespace crystal": [],
            "diamond": [],
            "plasma": [],
            "bananium": [],
            "plastic": [],
            "filename" : filename
        })
        for line in file:
            log = json.loads(line)
            if (type(log) is dict and log.get("cat") == "silo"):
                for key, value in material_totals.items():
                    material_totals[key] = 0
                    events[current_file][key].append({"timestamp":log.get("ts"), "total":0})
                continue
            raw_materials = log[3].get("raw_materials", "")
            if not raw_materials:
                continue
            for raw_material in raw_materials.split(", "):
                quantity, material = raw_material.split(" ", 1)
                material_totals[material] += int(quantity)
                events[current_file][material].append({"timestamp":float(log[4].get("timestamp")) - float(events[current_file][material][0]["timestamp"]), "total":int(material_totals[material])})
        for key, value in material_totals.items():
            events[current_file][key][0]["timestamp"] = "0"
        current_file += 1

for key, value in material_totals.items():
    dataframes = {}
    for round in events:
        df = pd.DataFrame(data=round[key])
        df["timestamp"] = pd.to_datetime(df["timestamp"], unit='s')
        df.index = df["timestamp"]
        df.sort_index()
        dataframes[round["filename"]] = df
    fig = go.Figure()
    for i in dataframes:
        fig = fig.add_trace(go.Scatter(x = dataframes[i]["timestamp"],
                                    y = dataframes[i]["total"], 
                                    name = i))
    fig.update_layout(title = key)
    fig.show()
```
 For example taking some recent Terry rounds of the last few days at peek times generates something like this for iron:
image

**Comment by ArcaneMusic** [comment]
Not dead just yet, spent this weekend working on a graphing utility for breaking down specific rounds similar to gamer's I just couldn't his to compile.
image
Working on verifying the results as an average, but the whole shows that that on average most of these observations appear consistent. I'll add this mineral script to the tool folder alongside this PR so we have it for future work as well.
image

**Comment by ArcaneMusic** [comment]
So: Good news bad news. The good news is I'm fairly confident that the changes made in this PR aren't noticeable enough to have a statistically significant negative impact on long term mineral balance, and therefore it should be alright to undraft.

The other good news is that the graphing script I wrote is bussin (sic) and works fairly well, so other people (largely me) and use it as a tool to look at mineral impact on rounds where we're screwing around with mineral balance.

The last good news is that the other PR by JohnFulpWillard covered the remaining missing defines for minerals from my last PR, so this should honestly be up to par for this specific change without going off the rails, which is word that can/should be done in another PR since this PR is just about tidied up from the looks of it.

The bad news is that this change really, doesn't impact mineral balance in a way that will actually matter to 99% of players. The impact that increasing minor mineral costs could have is looking at only like... 5% more usage per round of all minerals, maximum. We're talking like 4 sheets of iron and glass a round, and even less from the actual "rare" minerals I mentioned in the PR body. I've been mulling this over for a few weeks while I've been out on another work trip, and it's really hammered home that we could use a new approach to mineral design as a whole, and it's less to do with the costs (Although the availability of some items from the autolathe don't necessarily help matters) and more to do with the obscenely high amount that gets brought into the round that disables minerals from having any unique design choices to be made with them. That'll need to be looked at in another PR however.

For now, I'm opening this back up to review and going back to the drawing board for a future design doc.

**Comment by LemonInTheDark** [comment]
wag
